### ISSUE TITLE
Change behavior of Sync item count/limits 

### ISSUE THREAD

**Issue by mcackay** [phantomfish/byline#18]
Please correct me if my misunderstanding is wrong, but it appears that the "Sync X Items" feature loads the last X items from Feedly, ordered by item date descending. The effect of this being that if you have "Sync 1,000" items set - but some unread articles are in the 1,000 - 10,000 items range (according to Feedly) then they don't download. This then artificially reduces the number of 'unread items' and makes you feel like you have read everything .. but in fact are missing things.

I regularly get busy on my news feed (which is 500-1000 items per day) and then get a backlog which i try to catch up on over time. I also want to read some articles in my news feed when back at a computer, or in effect i'm "holding" them for use in my organisation newsletter/etc  (which might be better replaced by tagging) ... but the net effect is i have a lot of old unread items.

I suggest a change in methodology, to better suit more users. I assert the primary goals should be to:
* Always have the full unread list of articles syncing down to your device
* Have a certain number of 'saved' items on your device (which should include "All")
* Provide the user a way to access articles which were recently read, so you can mark as unread if read by mistake, or just want to revisit them.  For this point, I assert the 'recently read' list should be retained for a certain number of _days_ rather than items.

So I don't know the full Feedly API capabilities, but I'm hoping it would be possible to:

* Set a limit to the total number of UNREAD items you wish to sync down in Byline preferences. In a perfect world this would include the option 'All' -- but would be impacted by the 'Free Space' protection preference.
* Optionally limit the 'Saved for Later' count to X; but it should default to "All" [We did save it to read later, right... thus we want it on our device].
* Sync the status of an article as read/unread back up to the Feedly cloud with every sync.
* But retain the unread article in the device cache for X days _from the date marked as read_. 
* Purge items older than X days that have been read, to reduce the on-device cache size.

Once a sync occurs, if you have 'Hide Read' checked then you'd expect the article list view to not include them.  But it would be nice to have a button / pseudo-list item / etc called "Display recently read" ... which showed you the cached-but-read items in order by time read.

In general I also expect this would improve Byline performance, by allowing all read items to be purged after X days (e.g. 2 days) ... and so rather than retaining 10,000 items which 99% of us will never look at again ... you're only retaining 200-2000 or so which were recently viewed. Currently the database scan of the 10,000 cached items puts a real groan on performance for Byline ... but it's the only way to ensure I get an almost-full sync with the unread items I keep on Feedly.

**Comment by phantomfish** [comment]
Yes, you're correct about how this works now, and the limitations of the current model.

The problem is that if Byline only pulls unread items from Feedly, you won't get any items that have been read on other devices. I have to pull read as well as unread items, although not necessarily in the same call.

I could pull a large number of unread items and a smaller number of read items in separate calls to the API, but this isn't as efficient and would give the misleading impression that you have all the items (read and unread) until the date of the oldest unread item, when in fact many more recent read items may be missing.

I'm leaving this open as it's worth thinking about and debating....
